devicetypes/smartthings/springs-window-fashions-shade.src/springs-window-fashions-shade.groovy

```diff
@@ -266,4 +261,29 @@ def refresh() {
             zwave.switchMultilevelV1.switchMultilevelGet().format(),
             zwave.batteryV1.batteryGet().format()
     ], 1500)
+}
+
+def levelChangeFollowUp(expectedLevel) {
+    state.expectedValue = expectedLevel
+    state.levelChecks = 0
+    runIn(5, "checkLevel")
+}
+
+def checkLevelReport(value) {
+    if (state.expectedValue) {
+        if ((state.expectedValue == 99 && value > 99) ||
+                state.expectedValue == value) {
+            unschedule("checkLevel")
+        }
+    }
+}
+
+def checkLevel() {
+    if (state.levelChecks != null && state.levelChecks < 5) {
+        state.levelChecks += 1
+        runIn(5, "checkLevel")
+        response(zwave.switchMultilevelV1.switchMultilevelGet())
```

Comment:
I believe that you are actually going to need to pass this to `sendHubCommand`, otherwise nothing will actually get done with it (`response` just wraps the argument in a `HubAction` object).

Reply:
I think you might be right